src/OpenTelemetry.Instrumentation.AWS/Implementation/Metrics/AWSHistogram.cs

```diff
@@ -9,11 +10,25 @@ namespace OpenTelemetry.Instrumentation.AWS.Implementation.Metrics;
 internal sealed class AWSHistogram<T> : Histogram<T>
     where T : struct
 {
+    private static readonly ConcurrentDictionary<string, System.Diagnostics.Metrics.Histogram<T>> HistogramsDictionary
```

Comment:
naive question - any concern with effectively caching Histograms, Meters, etc indefinitely?   Looking to confirm it is necessary to clean up any old objects.

Reply:
I don't think there are major side effects other than the memory overhead considering that other instrumentations packages do reuse the instruments too (ex:https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInMetricsListener.cs#L36) 
though we do store more than just one histogram and meter.

For cleaning up old objects, we do create one meter per service the first time it is used and around a dozen metrics instruments each, the names of these instruments are constants that doesn't include any user defined value, so there will always be a chance that the old objects are needed later.